Update Product.php

Shoprunners opened the pull request:
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Make it possible to create a product via webservice in an empty category
| Type?             | bug fix
| Category?         | WS
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #14903
| Related PRs       | 
| How to test?      | Follow the instructions in the ussue
| Possible impacts? | unknown


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

Shoprunners committed:
Update Product.php

TomLorenzi commented on classes/Product.php:
```diff
@@ -7070,7 +7070,7 @@ public function setWsPositionInCategory($position)
         );
 
-        if ($position > count($result)) {
+        if ($position > count($result) && count($result) > 0) {
```
You should avoid repetting the same count twice, on big array it will take some times.

And here if your position is 5 in an empty category, you should still get an error that you would skip here

Shoprunners replied:
Good point,  I changed it.

TomLorenzi commented on classes/Product.php:
```diff
@@ -7087,7 +7087,9 @@ public function setWsPositionInCategory($position)
         array_unshift($result, null);
         foreach ($result as &$value) {
-            $value = $value['id_product'];
+            if($value != null) {
```
Just move the array unshift after the `foreach` would avoid getting for logic in here

Shoprunners committed:
Performance improvements

Performance improvement of checking valid position by counting the resultset only once
